lib/shared-sku-app/src/Editor/Field.tsx

```diff
@@ -0,0 +1,163 @@
+  isDisabled: DisabledPredicateFn;
+}
+
+interface State {
+  value: string[] | string;
+  valid: boolean;
+  editingDisabled: boolean;
+}
+
+const styles = {
+  sortable: css({
+    marginBottom: tokens.spacingM
+  }),
+  container: css({
+    display: 'flex'
+  }),
+  logo: css({
+    display: 'block',
+    width: '30px',
+    height: '30px',
+    marginRight: tokens.spacingM
+  })
+};
+
+function getEmptyValue(sdk: FieldExtensionSDK) {
+  const isArray = sdk.field.type === 'Array';
+  return isArray ? [] : '';
+}
+
+export default class Field extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const value = props.sdk.field.getValue();
+    const isValidFieldType =
+      typeof value === 'string' ||
+      (Array.isArray(value) && value.every(v => typeof v === 'string'));
+
+    // `valid` is `true` if the app can render/write the value safely.
+    // If for example there is an object stored we don't want to override
+    // it without a user explicitly telling us to do so.
+    const valid = isNil(value) || isValidFieldType;
+
+    this.state = {
+      value: value || getEmptyValue(this.props.sdk),
+      valid,
+      editingDisabled: false
+    };
+  }
+
+  componentDidMount() {
+    this.props.sdk.window.startAutoResizer();
+
+    // Handle external changes (e.g. when multiple authors are working on the same entry).
+    this.props.sdk.field.onValueChanged((value?: string[] | string) => {
+      this.setState({ value: value || getEmptyValue(this.props.sdk) });
+    });
+
+    // Disable editing (e.g. when field is not editable due to R&P).
+    this.props.sdk.field.onIsDisabledChanged((editingDisabled: boolean) => {
+      this.setState({ editingDisabled });
+    });
+  }
+
+  updateStateValue = async (value: string[] | string) => {
+    this.setState({ value });
+    if (value.length > 0) {
+      await this.props.sdk.field.setValue(value);
+    } else {
+      await this.props.sdk.field.removeValue();
+    }
+  };
+
+  onDialogOpen = async () => {
+    const currentValue = this.state.value;
+    const config = this.props.sdk.parameters.installation;
+    const isArray = this.props.sdk.field.type === 'Array';
+    if (isArray) {
+      const result = await this.props.openDialog(this.props.sdk, currentValue, {
+        ...config,
+        fieldValue: this.props.sdk.field.getValue(),
+        fieldType: this.props.sdk.field.type
+      });
+      if (result.length) {
+        await this.updateStateValue(result);
+      }
+    } else {
+      const result = await this.props.openDialog(this.props.sdk, currentValue, {
+        ...config,
+        fieldValue: this.props.sdk.field.getValue(),
+        fieldType: this.props.sdk.field.type
+      });
+      const product = get(result, ['0'], null);
+      if (product) {
+        await this.updateStateValue(product);
+      }
+    }
+  };
+
+  render = () => {
+    const { value, valid, editingDisabled } = this.state;
+
+    if (!valid) {
```

Comment:
This can be safely removed, field types already enforce the value is correct

Reply:
Removed